### ISSUE TITLE
Unable to get job queue to accept jobs

### ISSUE THREAD

**Issue by zacharynevin** [hunterloftis/jackrabbit#8]
Hello,

After following your example, I have the following files:

- `server.js`. Node simple web server.
- `worker.js`. Worker.

What I want to do is have a job queue where in a web request I can push data to a job.

In the worker.js file, I do the following:
````
// worker.js
var http = require('http');
var throng = require('throng');
var jackrabbit = require('jackrabbit')
 // want to use "worker: node worker.js" in heroku Procfile
var queue = jackrabbit(process.env.CLOUDAMQP_URL || 'amqp://localhost')

http.globalAgent.maxSockets = Infinity;

var start = function() {
  queue.on('connected', function() {
    queue.create('myJob', { prefetch: 5 }, function() {
        queue.handle('myJob', function(data, ack) {
            console.log(data)
            console.log("Job completed!")
            ack()
        })
    })
  })
}

throng(start, { 
  workers: 1,
  lifetime: Infinity,
  grace: 4000
})
````

Now I want a situation where I can push data to a job from the web application. So I in a middleware function I have,

````
// middleware.js
var jackrabbit = require('jackrabbit')
var queue = jackrabbit(process.env.CLOUDAMQP_URL || 'amqp://localhost')

app.get('/example', function(req, res, next) {
   queue.publish('job_name', { data: 'my_data' })
   res.send(200, { data: 'my_data' })
})
````
On development, I start `worker.js` and `app.js` in separate terminal windows. When I call the `/example` method, I expect to see the worker terminal window show the relevant console logs. However, the request just hangs.

I feel like there is something really fundamental I am missing here in terms of my understanding of jackrabbit and ampq. Any help on this would be greatly appreciated.

**Comment by haircuttedfreak** [comment]
It looks like the newest version on NPM (3.0.8) has a bug.  There is a hard coded expiration on line 77 of queue.js, which mean all the message are expiring in one second by default with no way to override.  I think the intention was to pass the options.messageTtl  from the Queue constructor to the expiration property.

Unfortunately it looks like the 3.0.8 version of the code wasn't pushed up to Github.  The latest master on Github is 3.07, which doesn't have the issue.  Therefore I can't fork and submit a pull request.

@hunterloftis it would be great if you could push up the 3.0.8 code with a bug fix.

Otherwise the best solution for me was to use version 3.0.7 in my package.json file.

**Comment by hunterloftis** [comment]
@haircuttedfreak thanks for the sleuthing! Pushed.

**Comment by zacharynevin** [comment]
Setting the `messageTtl` on the `create` function didn't really work for me. The way I am understanding the process is this:
- Start `worker.js`, where the queue is created and the handler function set.
- Start `server.js`, running the webserver. When I need to push data to the queue, I simply connect to RabbitMQ through Jackrabbit and call the `publish` method with the queue name and data, without needing to create the queue first.

**Comment by haircuttedfreak** [comment]
I submitted pull request with the bug fix.  You can see my fork at https://github.com/haircuttedfreak/jackrabbit.  Give that a shot and see if it works for you.




I do believe you will need to create the queue first like in the example code.  It’s really a “create the queue if it doesn’t already exist” assertion.  So there’s not harm in calling it multiple times from multiple places.

**Comment by createproblem** [comment]
@hunterloftis Why is the pull request from @haircuttedfreak not merged?

This option is important. A hardcoded expiration is really bad, I become crazy to figure out where the problem is...

**Comment by hunterloftis** [comment]
Thanks for all the help @haircuttedfreak, just published your PR in 3.0.9.
